src/effects/ladspa/LadspaEffect.cpp

```diff
@@ -1803,3 +1802,16 @@ void LadspaEffect::Validator::RefreshControls()
       mFields[p]->SetValue(fieldText);
    }
 }
+
+void LadspaEffect::Validator::UpdateControl(int index, float value)
+{
+   auto& controls = mSettings.controls;
+
+   assert(index < static_cast<int>(controls.size()));
+
+   if (std::abs(controls[index] - value) < std::numeric_limits<float>::epsilon())
```

Comment:
Is this tolerance test necessary, and if so, is it correct?

"Returns the machine epsilon, that is, the difference between 1.0 and the next value representable by the floating-point type T."

But possibly some control is scaled to an interval of small fractional values.

All the Ladspa sliders are graded to 1000 steps.  Maybe the epsilon should be 1/1000 times the difference of maximum and minimum values of the control.

Reply:
Or rather, make that 1/2000 -- one-half of the interval.

Reply:
After careful consideration I don't think that using sliders step as a delta is correct. For LADSPA machine epsilon appears to be a good approximation, but for consistency I will define it to match the one LV2. For LV2 epsilon should be larger (but still - it is unrelated to amount of steps we define for sliders), because we allow text input in the plain UI (and the epsilon is only used in plain UI). 

For VST3 I know plugins, that require epsilon around 10^-6